src/coreclr/nativeaot/Runtime/wasm/wasi.cpp

```diff
@@ -0,0 +1,63 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "wasi.h"
+#include <signal.h>
+#include <sys/mman.h>
+#include <malloc.h>
+#include "CommonTypes.h"
+#include "../CommonMacros.h"
+#include <string.h>
+
+int getrlimit (int resource_id, struct rlimit * ret_rlimit)
+{
+  switch(resource_id)
+  {
+    case RLIMIT_AS :
+      // TODO-LLVM: how to implement rlimits for Wasi
+      // The gc wants to know this in GetRestrictedPhysicalMemoryLimit.  Will try with an arbritrary value of 1GB
+      ret_rlimit->rlim_cur = 1024 * 1024 * 1024;
+      ret_rlimit->rlim_max = 1024 * 1024 * 1024;
```

Comment:
How is the Browser path implemented for this?

Reply:
Emscripten's muslc implements it with syscalls `SYS_prlimit64` and `SYS_getrlimit` : https://github.com/emscripten-core/emscripten/blob/90d76db21200d42f5536bc3a01a22303e398d8b4/system/lib/libc/musl/src/misc/getrlimit.c#L7-L24

Wasi SDK's does not include that file, https://github.com/WebAssembly/wasi-libc/blob/1dfe5c302d1c5ab621f7abf04620fae92700fd22/Makefile#L88-L101

Reply:
> Shall I change the limits to ~0ULL ?

Seems reasonable.